### DISCUSSION TITLE
Empty array set to NULL after request

### DISCUSSION THREAD

**Discussion by artworkad** [livewire/livewire#3840] [7. Bugs (archived)] [answered]
Say there is following model

```
class Test extends Model
{
    protected $fillable = [
        'list'
    ];
}

```
In mount of a livewire component:

    $model = new Test();
    $model->list = [];

In view:

    {{ var_dump($model->list) }}

prints

    array(0) { }

But on a subsequent request it becomes 

    NULL

This is new in 2.6.

**Comment by joshhanley** [comment]
@artjomzab can you confirm, do you have rules setup? If so, can you please add them to your example.

If not, you need to make sure you have rules. Any properties you want to persist when using model binding, need to have a rule defined (even if it's just 'nullable').

See in the docs here https://laravel-livewire.com/docs/2.x/properties#binding-models

> Note: For this to work, you have a validation entry in the $rules property for any model attributes you want to bind to. Otherwise, an error will be thrown.

Although you are not binding to it, the only way Livewire keeps state is by sending the data to the front end, and the rules array is a whitelist for what data to send.

Hope this helps!

**Reply by artworkad** [reply]
Thank's for your feedback @joshhanley. Rules are setup, nonetheless list property becomes NULL:

    public Test $test;

    public function mount() {
        $this->test = new Test();
        $this->test->list = [];
    }

    public function rules(): array
    {
        return [
            'test.list' => 'required'
        ];
    }

**Comment by basepack** [comment]
Having exact same behavior on my save method. 

Given this Component:
```
    public $rules = [ 'model.property_name' => 'array' ];

    public function hydrate(){
        // prevent livewire to set these to null
        $this->model->property_name ??= [];
    }
    
    public function save()
    {
        //
    }
```

- in the $rules an 'array' property:
- hit a save button
- property is empty
- I added that hydrate method to fix this from happening.

@joshhanley any idea what causes this? As far as I know this has been happening for at least half a year now.

The exact error comes from the blade view in my case, because `@foreach` cannot loop `null`:
```
ErrorException
foreach() argument must be of type array|object, null given (View: /app/resources/views/livewire/admin/model.blade.php)
```

**Reply by joshhanley** [reply]
@basepack hmm I'm not sure. So did this start happening in 2.6 as reported above or has it been happening before that?

I could understand if it changed in 2.6, as we added the deep model binding feature, which could have some bugs.
But if it was already happening before 2.6, then we would have written the deep model bind feature to match how it worked before.

**Reply by jeffchown** [reply]
@joshhanley Same thing is happening to me for all forms since updating to 2.6 (I also updated the server involved to PHP 8.0 at the same time).  

Using the same workaround as @basepack

**Reply by joshhanley** [reply]
@jeffchown ok great, thanks for confirming re 2.6 🙂 looks like it could be an issue with deep model binding. When I get a chance, I will have a look to see if I can track it down.

**Comment by joshhanley** [comment, marked as answer]
@artjomzab, @basepack and @jeffchown I've tracked the issue down and submitted a PR #4135 with the fix.

Hope this helps!

**Reply by basepack** [reply]
Great Josh! 👍

**Reply by joshhanley** [reply]
No worries! 🙂
